pyairtable/api/types.py

```diff
@@ -3,9 +3,17 @@
 and return values to various pyAirtable methods.
 """
 from functools import lru_cache
-from typing import Any, Dict, List, Optional, Type, TypeVar, Union, cast
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Type, TypeVar, Union, cast
+
+if TYPE_CHECKING:  # mypy really does not like this conditional import.
+    import pydantic
+else:
+    # Pydantic v2 broke a bunch of stuff. Luckily they provide a built-in v1.
+    try:
+        import pydantic.v1 as pydantic
+    except ImportError:
+        import pydantic
```

Comment:
Should we move this somewhere (maybe `pyairtable.utils` or a new module `pyairtable.compat`) so it only happens once, and then import from there in the rest of our code?

Reply:
I think this is a natural reaction to seeing duplicated code, but I feel in this case, the abstraction will just make it harder to follow what's going on.

Reply:
Actually, since you are adding more pedantic, it definitely makes sense to move this into one place. I will do it.

Reply:
Thanks! I think most library users are not going to peek under the hood at how we're using pydantic; they're just going to want to import `Base` and `Table` and get on with life. So if it makes it easier for us (and I think it will, given how many places I'm starting to use pydantic) it's worth the time.